ui/src/utils/stateConversion.js

```diff
@@ -0,0 +1,52 @@
+import { uniqueIdGen } from "./uniqueIdGenerator";
+
+export const serverToUIConversion = input => {
+  try {
+    // subpolicy stringify + give Id
+    for (let sp of input.Subpolicies) {
```

Comment:
Minor: It should be possible to use `for (const sp ...) instead of `for (let sp ...)` for all your `for` statements.

Side note: `const` in JS only gives exterior immutability (i.e. cannot reassign variable), but the interior state is still mutable. That also means pretty much all the values can be (and should be) declared as `const`.
Ref: https://ponyfoo.com/articles/const-variables-not-immutable